src/main/java/co/cask/gcp/gcs/GCSConfigHelper.java

```diff
@@ -25,17 +25,31 @@
 
   private static final String FILE_SYSTEM_PATH_PREFIX = "gs://";
 
-  private static final Pattern BUCKET_PATTERN = Pattern.compile("gs:\\/\\/(.*?)\\/.*");
-
   /**
    * Returns the given path as it is if it starts with {@link GCSConfigHelper#FILE_SYSTEM_PATH_PREFIX} else appends it
    * in the beginning.
    *
    * @param path the path
-   * @return the path with which starts with {@link GCSConfigHelper#FILE_SYSTEM_PATH_PREFIX}
+   * @return {@link URI} for the given path
+   * @throws IllegalArgumentException if the given path is not valid
    */
-  public static String getPath(String path) {
-    return path.startsWith(FILE_SYSTEM_PATH_PREFIX) ? path : FILE_SYSTEM_PATH_PREFIX + path;
+  public static URI getPath(String path) {
+    try {
+      URI uri = new URI(path);
+      if (uri.getScheme() != null && !uri.getScheme().equalsIgnoreCase("gs")) {
+        throw new URISyntaxException(path, String.format("Invalid scheme. Scheme must be '%s'.",
```

Comment:
I think for any of these exceptions, the message can be:
`Invalid path. The path must be of the form 'gs://<bucket>/path'.`